tests/integration/tracing/test_struct.py

```diff
@@ -0,0 +1,142 @@
+from collections.abc import Callable
+from typing import Generic
+
+from guppylang.decorator import guppy
+from guppylang.module import GuppyModule
+
+
+def test_create(validate, run_int_fn):
+    module = GuppyModule("module")
+
+    @guppy.struct(module)
+    class S:
+        x: int
+        y: int
+
+    @guppy.comptime(module)
+    def main(x: int) -> int:
+        s = S(x, 2)
+        return s.x + s.y
+
```

Comment:
Would this work and if so can it be a test?

```suggestion
    @guppy(module)
    def main(x: int) -> int:
        s = comptime(S(x, 2))
        return s.x + s.y

```

Reply:
Doesn't work yet, but on my TODO list for another PR